src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JenkinsAppApi.java

```diff
@@ -101,4 +127,17 @@ private ResponseEntity handleResponseBody(
         return objectMapper.readValue(
                 body.bytes(), objectMapper.getTypeFactory().constructType(responseClass));
     }
+
+    private String wrapInJwt(final JenkinsAppRequest request, final String secret)
+            throws JsonProcessingException {
+        final String body = objectMapper.writeValueAsString(request);
+        Algorithm algorithm = Algorithm.HMAC256(secret);
+        return JWT.create()
+                .withIssuer("jenkins-plugin")
+                .withAudience("jenkins-forge-app")
+                .withIssuedAt(new Date())
+                .withExpiresAt(Date.from(Instant.now().plusSeconds(5 * 60)))
+                .withClaim("request_body_json", body)
+                .sign(algorithm);
+    }
```

Comment:
This is where the JWT for the request body is generated. It contains a claim `request_body_json` that contains the actual request as a JSON string (with escaped quotes).